packages/angular_devkit/build_angular/src/tools/esbuild/stylesheets/sass-language.ts

```diff
@@ -97,31 +103,49 @@ async function compileString(
             url,
             options: FileImporterWithRequestContextOptions,
           ): Promise<URL | null> => {
+            let resolved = resolutionCache.get(url);
```

Comment:
Suggestion: Similar to the comment thread in the previous PR, I wonder if we could separate out the caching logic to a different layer. Something like decorating or memoizing the `findFileUrl` function with the resolution cache? That would allow readers to think of this implementation as two functions, one which resolves URLs and another which caches them. `if (result.path)` can turn from a mutation to an early return which would make it much easier to follow IMHO.

Could arguably apply the same thing to `packageRootCache`, though that usage is much more localized so I'm less concerned about that one.